arelle/plugin/validate/ESEF/ESEF_Current/ValidateXbrlFinally.py

```diff
@@ -311,6 +313,14 @@ def checkFootnote(elt: ModelInlineFootnote | ModelResource, text: str) -> None:
                        "%(url)s: %(documentSets)s (Document files appear to be in multiple document sets)"),
                 modelObject=doc, documentSets=", ".join(sorted(ixdsDocDirs)), url=reportIncorrectlyPlacedInPackageRef)
         ixTargetUsage = val.authParam["ixTargetUsage"]
+        earliestTransformationRegistry = val.authParam.get("earliestTransformationRegistry","zzzz")
+        allowedTRnamespaces = set()
+        earliestTRnumber = ""
+        for key, ns in ixtNamespaces.items():
+            if ns == earliestTransformationRegistry:
+                earliestTRnumber = key[5:]
+            if ns >= earliestTransformationRegistry:
+                allowedTRnamespaces.add(ns)
```

Comment:
This is a little fragile. It would be better to explicitly handle None/unset `earliestTransformationRegistry` instead of using a `zzzz` magic string, and to use a regex to parse the date from the namespace value, raising a `ValueError` if it doesn't match the expected pattern.

Reply:
simplified by adding a TRnumber feature to FunctionIxt module next to the other relevant dicts and ditching zzzz